WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -173,7 +182,9 @@ private void unpackPostIntent(@NonNull Intent intent) {
             if (!hasPendingOrInProgressMediaUploadsForPost(post)) {
                 mPostUploadHandler.upload(post);
             } else {
-                sPostsWithPendingMedia.add(post);
+                List<MediaModel> pendingMedia = MediaUploadHandler.getPendingOrInProgressMediaUploadsForPost(post);
+                UploadingPost uploadingPost = new UploadingPost(post, pendingMedia);
```

Comment:
If we used the full count of media items inside a post to calculate the percentage, we'd be showing some strange results for posts with media uploaded a long time before. Say we upload a post and it has 4 media, and then we edit it much later (days, weeks) and add 1 media, I don't think it's clear at all to show a progress bar starting at `80%` right away.

I agree the behavior is not ideal when the uploads are in progress, and we open the editor and see one complete, and then go back to the list. We could do a 'sessional' approach, where we remember all the uploads since we first queued the post for upload, so opening the editor and media completing won't affect the total count of uploading media.